### DISCUSSION TITLE
Series with more than 100 episodes

### DISCUSSION THREAD

**Discussion by ray2301** [trix7777/m2strm#12] [General]
Hi,
Thanks for this code, I use it daily. I actually learned how to modify it for my provider and built my first exe ever :)
//i deleted everything else because i've found the problem and it wasn't your code :)

**Comment by trix7777** [comment]
Please upload your modified version as a fork! =) I will happily merge changes to the code if possible.

Did you manage to get it to work with eps 100+? I have never had a show with that many eps to test it on.

**Comment by trix7777** [comment]
Moving this topic to Discussions

**Comment by ray2301** [comment]
oh it always worked with more than 100 episodes, it was the merged playlist i used that cut episodes above E99 to E## and i haven't noticed it before I actually wrote this (the playlist has more than 10000 shows so that's a lot of lines of text). actually i didn't realize untill i learned what (\\d+) does and that there's no error in your code :) then i checked my playlist.
It's not really a fork, I haven't changed anything. I just needed to be able to change how m2strm recognizes tv shows because my links do not contain "/shows" for tv shows, but i can differentiate them from movies and tv with another keyword. I just didn't know how to build this thing so I managed to learn just because of that :)

it would be cool if we could be able to change those keywords right in the config file but i don't know how to do that.
edit2: something like recognition of unwanted group files could also check if a group name is XY and set that to tv shows. some playlists have "TV VOD" or something like that that differentiates them from movies and tv, some have different links but could be differentiated by a part of a link and so on...
it could also check the whole first line for text if any other differentiator exists in the whole 1st line and the other differentiator could be the 2nd line (link).
differentiator for shows could maybe also be S(\\d+)E(\\d+) when there are no other differentiators and god knows i tried to do that one, but no way :)

so this could work on much more combinations od playlists if that could be set in a configuration file instead of changing the code completely for a certain part. because that is its biggest problem with compatibility with more playlist types of m3us.

i wish i could manage that, but i'm probably too old to start now and i have no background in coding other than snooping in the code to find something i know i'm searching for and changing it and googling on how to do something or what something means. if i had to deal with regex all alone i'd probably lose a lifetime :) 
(i've never used the word differentiate so much before)

**Reply by trix7777** [reply]
I would probably implement something like what you propose above if my brain would work like it did before I had a mental breakdown after getting canned and then got covid and now post-covid. Now I'm more or less at a loss and, looking at the code now, I can't believe I wrote it, though I have not lost hope of getting that part of my brain back.

Regex is powerful and fun =) I based this app on another one (M3U_to_STRM) written in VB.NET but in the end I re-wrote it from scratch to get much more functions from it. I did however fork it at first (with permissions from the original author) but after a while it got to be a totally different code, now in C#. I did not know either of those two code languages previously, but used google and alot of regex101 to get the work done =)

**Comment by ray2301** [comment]
i feel for you, i had cancer, the break-up came later and that kinda messed with me all together. cancer messed my body a lot, break-up messed my head and how i feel about new relationships (and all together messed with how i feel about people). many years passed, now i'm almost 40 and home a lot so i play with things like this when i get some idea like "could i export vod from m3u as strm files to import them to my library?". then i find this, then i don't understand it, then i learn how to use it, then something isn't working as expected, then i try to understand what and how to fix it. and time passes, that kind of becomes my little personal project  :) i can still learn but i can't concentrate like i could before so everything takes what seems to me like a really long time to figure out.

i actually tried M3U_to_STRM and even looked at the code, but yours seemed more complete and understandable. i still didn't look at how you did the "recognition of unwanted group files" in the code but i said to myself i'll try to decipher that to try to understand how that works, kinda helps me work my brain out even though it doesn't help in real life :) i just hope my beard doesn't outgrow my face before i get it.
thanks again for this!
